### DISCUSSION TITLE
Licensing issues on CI

### DISCUSSION THREAD

**Discussion by miquecg** [ibm-messaging/mq-container#556] [Q&A]
Hi,

Would the following scenario violate Licence Agreement?

> An application and all its components, including IBM MQ, running on containers as part of a CI pipeline shared among members of a team.

To my understanding, since this is for development purposes only, namely testing, it should be fine.

Note that we are referring to IBM MQ images published on `icr.io` registry (old ones still exist on Docker Hub) and also `arm64` compatible images built from this repo.

Thanks!

**Comment by arthurbarr** [comment]
MQ Advanced for Developers is licensed for a single user, so whilst an image can be hosted in a shared repository, the license only permits each deployment to be used by a single user for Development purposes. For the shared environment described here, the IBM MQ Advanced for Non-Production Environment entitlement is more appropriate.

**Reply by miquecg** [reply]
Hi @arthurbarr, thanks for your reply.

What does the Licence consider as _deployment_? Is it the IBM MQ installation?

For that matter, each IBM MQ installation serves a single developer running tests on a pipeline and it only exists in the context of that execution. Every provisioned container and volume is ephemeral and cleaned upon finish. The only shared resource is the underlying OS on a virtual machine.

**Comment by arthurbarr** [comment]
I think you might be better off sending an email to askmessaging@uk.ibm.com.  The product management and legal teams do not generally engage on GitHub, and you'll need to go through them if you want to discuss the details of the license terms.

**Reply by miquecg** [reply]
Oks, thanks 🙂
